libs/langchain-community/src/chat_models/premai.ts

```diff
@@ -0,0 +1,471 @@
+import {
+  AIMessage,
+  AIMessageChunk,
+  type BaseMessage,
+  ChatMessage,
+  ChatMessageChunk,
+  HumanMessageChunk,
+} from "@langchain/core/messages";
+import {
+  type BaseLanguageModelCallOptions,
+  TokenUsage,
+} from "@langchain/core/language_models/base";
+
+import { CallbackManagerForLLMRun } from "@langchain/core/callbacks/manager";
+import {
+  type BaseChatModelParams,
+  BaseChatModel,
+} from "@langchain/core/language_models/chat_models";
+
+import Prem, {
+  ChatCompletionStreamingCompletionData,
+  CreateChatCompletionRequest,
+  CreateChatCompletionResponse,
+} from "@premai/prem-sdk";
+import { getEnvironmentVariable } from "@langchain/core/utils/env";
+import {
+  ChatGeneration,
+  ChatGenerationChunk,
+  ChatResult,
+} from "@langchain/core/outputs";
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { AxiosRequestConfig } from "axios";
```

Comment:
This is kinda incidental, can we make this a peer dep and install a version as part of setup?